validator/db/proposal_history.go

```diff
@@ -0,0 +1,62 @@
+package db
+
+import (
+	"github.com/boltdb/bolt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	slashpb "github.com/prysmaticlabs/prysm/proto/slashing"
+)
+
+func unmarshalProposalHistory(enc []byte) (*slashpb.ProposalHistory, error) {
+	history := &slashpb.ProposalHistory{}
+	err := proto.Unmarshal(enc, history)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal encoding")
+	}
+	return history, nil
+}
+
+// ProposalHistory accepts a validator public key and returns the corresponding proposal history.
+// Returns nil if there is no proposal history for the validator.
+func (db *Store) ProposalHistory(pubKey []byte) (*slashpb.ProposalHistory, error) {
+	var err error
+	var proposalHistory *slashpb.ProposalHistory
+	err = db.view(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(historicProposalsBucket)
+		enc := bucket.Get(pubKey)
```

Comment:
Similar to above; a `nil` check would be safer (an ditto for other fetches throughout the code).